pkg/translate/translate.go

```diff
@@ -0,0 +1,57 @@
+package translate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/expect-digital/translate/pkg/convert"
+	"github.com/expect-digital/translate/pkg/model"
+	pb "github.com/expect-digital/translate/pkg/server/translate/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type TranslateServiceServer struct {
+	pb.UnimplementedTranslateServiceServer
+}
+
+func New() *TranslateServiceServer {
+	return new(TranslateServiceServer)
+}
+
+func (t *TranslateServiceServer) UploadTranslationFile(
+	ctx context.Context,
+	req *pb.UploadTranslationFileRequest,
+) (*pb.UploadTranslationFileResponse, error) {
+	if len(req.GetTranslationId()) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "Missing translation_id")
+	}
+
+	var (
+		messages model.Messages
+		err      error
+	)
+
+	switch req.GetType() {
+	default:
+		return nil, status.Errorf(codes.Internal, "")
+	case pb.Type_TYPE_UNSPECIFIED:
+		return nil, status.Errorf(codes.InvalidArgument, "Type is missing")
+	case pb.Type_NG_LOCALISE:
+		messages, err = convert.FromNgJson([]byte(req.Data))
+	case pb.Type_NGX_TRANSLATE:
+		messages, err = convert.FromNgxTranslate([]byte(req.Data))
+	case pb.Type_GO:
+		messages, err = convert.FromGo([]byte(req.Data))
+	}
```

Comment:
Functions are first-class citizens in golang. Define `type From = func ([]byte) (model.Messages, error)` in the `convert` package.

```suggestion

	var from convert.From
	
	switch v := req.GetType(); v {
	default:
		return nil, status.Errorf(codes.Unimplemented, "Type '%s' is unimplemented", pb.Type_name[v])
	case pb.Type_TYPE_UNSPECIFIED:
		return nil, status.Errorf(codes.InvalidArgument, "Type is missing")
	case pb.Type_NG_LOCALISE:
		from = convert.FromNgJson
	case pb.Type_NGX_TRANSLATE:
		from = convert.FromNgxTranslate
	case pb.Type_GO:
		from = convert.FromGo
	}
	
	messages, err := from([]byte(req.Data)
	// ...
```